Templates/VSIX/Project Templates/cpp-winui-template/WinUI3TemplateCpp/App.xaml.cpp

```diff
@@ -6,32 +6,33 @@
 #include "App.xaml.h"
 #include "MainWindow.xaml.h"
 
-using namespace winrt;
-using namespace Microsoft::UI::Xaml;
-using namespace Microsoft::UI::Xaml::Controls;
-using namespace Microsoft::UI::Xaml::Navigation;
-using namespace Windows::Foundation;
-using namespace $safeprojectname$;
-using namespace $safeprojectname$::implementation;
+namespace winrt
```

Comment:
@oldnewthing - the existing "slurk everything into the global namespace in a file" is the style in the OS samples. Out of curiosity, is there a meaningful difference?

Reply:
@jonwis There are two issues here, and the OS samples repo is wrong on one count, possibly both. Fixing them is something I do only opportunistically, though.

First problem is the way we "slurk" the namespaces. Should be

```c++
using namespace winrt;
using namespace winrt::Windows::Foundation;
//              ^^^^^^^
```

If you leave off the `winrt::` prefix on `winrt::Windows::Foundation`, then you get a compile error if you try to use both C++/WinRT and WRL (which many people do).

```c++
#include <winrt/Microsoft.UI.Xaml.h>
#include <wrl/client.h>

using namespace winrt;
using namespace Microsoft::UI::Xaml; // error: "Microsoft" is ambiguous
// could be winrt::Microsoft or ::Microsoft
```

The second problem is whether we should be dumping everything into the global namespace, or consolidating in a `winrt` namespace:

```c++
// This is not controversial
namespace winrt
{
    using namespace Windows::Foundation;
}

// this part is
using namespace winrt;
```

For code that straddles multiple worlds, dumping everything into the global namespace is a bad idea because you want to be clearer about which `DispatcherQueue` you are talking about.

* `winrt::Windows::System::DispatcherQueue`
* `Windows:System::DispatcherQueue` (C++/CX dumps in the global namespace, alas)
* `ABI:Windows::System::DispatcherQueue`

The OS samples get away with `using namespace winrt;` because they live in only one world (C++/WinRT).

The WindowsAppSDK samples might or might not fall into the same category. If they need to juggle multiple projections and namespaces, then consolidating into a `winrt` non-global namespace may be better.

```c++
winrt::DispatcherQueue queue = blah blah; // Aha, this is the C++/WinRT version
ABI::DispatcherQueue* abi_queue = winrt::get_abi(queue); // convert to raw ABI
```
```

Reply:
To be conservative and allow for calling other namespaces with WinAppSDK, does it sound ok to keep the pattern here? (e.g. global winrt namespace)?